frontend/src/components/common/Filter/common/entities.ts

```diff
@@ -1,58 +1,301 @@
-import { ChangeEvent, Dispatch, SetStateAction } from "react";
-import { CellValue, Row } from "react-table";
+import { CellValue, FilterValue, Row } from "react-table";
```

Comment:
* Added discriminating unions for category filter configs and multi-panel panel configs.
* Moved entities from `useCategoryFilter` to here that now need be accessed by the individual category filter-specific util files.